### PR TITLE
nostr-rs-relay: init at 0.8.9 

### PR DESCRIPTION
**Pull request by jurraca**
###### Description of changes

nostr-rs-relay provides an implementation of a Nostr [relay](https://nostr.how/en/relays) written in Rust by [@scsibug](https://github.com/scsibug/).

###### Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [X] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [X] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by jurraca** [comment]
@scsibug and @jb55 if you want to be maintainers here, happy to add you.

File: pkgs/misc/nostr-rs-relay/default.nix [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+  stdenv,
+  fetchFromGitHub,
+  rustPlatform,
+  openssl,
+  pkg-config,
+  libiconv,
+  darwin,
+  protobuf,
+}:
+rustPlatform.buildRustPackage rec {
```
Comment (natsukium) [inline pkgs/misc/nostr-rs-relay/default.nix:12]:
```suggestion
{ lib
, stdenv
, fetchFromGitHub
, rustPlatform
, openssl
, pkg-config
, libiconv
, darwin
, protobuf
}:

rustPlatform.buildRustPackage rec {
```

**Comment by natsukium** [comment]
It looks fine.
`pkgs/servers/nostr-rs-relay` seems more appropriate than `pkgs/misc/nostr-rs-relay`.

Then would you please squash the commits to `maintainers: add jurraca` and `nostr-rs-relay: init at 0.8.9`?

**Comment by natsukium** [comment]
Result of `nixpkgs-review pr 227946` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>nostr-rs-relay</li>
  </ul>
</details>

**Comment by jurraca** [comment]
many thanks for the review @natsukium !

File: pkgs/servers/nostr-rs-relay/default.nix [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+, fetchFromGitHub
+, rustPlatform
+, openssl
+, pkg-config
+, libiconv
+, darwin
+, protobuf
+,
+}:
+rustPlatform.buildRustPackage rec {
```
Comment (natsukium) [inline pkgs/servers/nostr-rs-relay/default.nix:12]:
```suggestion
}:

rustPlatform.buildRustPackage rec {
```

**Review by natsukium — COMMENTED** [review]
Thanks for your response.
The commit unit is perfect. Could you please change the message to `nostr-rs-relay: init at 0.8.9` and `maintainers: add jurraca`?
It would be nice if "maintainers: add jurraca" is the first commit.

File: pkgs/servers/nostr-rs-relay/default.nix (hunk 2 above)
Comment (jurraca) [reply pkgs/servers/nostr-rs-relay/default.nix]: thanks, I missed the message detail, makes sense.

File: pkgs/servers/nostr-rs-relay/default.nix [hunk 3]
```diff
@@ -0,0 +1,45 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, rustPlatform
+, openssl
+, pkg-config
+, libiconv
+, darwin
+, protobuf
+,
```
Comment (natsukium) [inline pkgs/servers/nostr-rs-relay/default.nix:10]:
```suggestion
```
Except for this, it looks fine.

**Review by natsukium — APPROVED** [review]

File: pkgs/servers/nostr-rs-relay/default.nix [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+    protobuf
+  ];
+
+  meta = with lib; {
+    description = "Nostr relay written in Rust";
+    homepage = "https://sr.ht/~gheartsfield/nostr-rs-relay/";
+    changelog = "https://github.com/scsibug/nostr-rs-relay/releases/tag/${version}";
+    maintainers = with maintainers; [ jurraca ];
+    license = licenses.mit;
+    platforms = platforms.linux ++ platforms.darwin;
```
Comment (SuperSandro2000) [inline pkgs/servers/nostr-rs-relay/default.nix:42]:
```suggestion
```
that is the default for buildRustPackage

**Pull request merged**